### PR TITLE
Fix weak spacing being ignored unconditionally in `math.lr`

### PR DESCRIPTION
**Pull request by mkorje**
Fixes a bug in #2950: weak spacing fragments were being removed unconditionally, regardless of whether there was actually an opening/closing delimiter in the arguments passed.

| Before | After |
| - | - |
| old | new |

Result of the Typst code below before and after this change.

```typ
--- math-lr-weak-spacing ---
// Test ignoring weak spacing immediately after the opening
// and immediately before the closing.
$ [#h(1em, weak: true)A(dif x, f(x) dif x)sum#h(1em, weak: true)] $
$ lr(\[#h(1em, weak: true)lr(A dif x, f(x) dif x\))sum#h(1em, weak:true)a) $
```

### CODE REVIEW

File: crates/typst-layout/src/math/lr.rs [hunk 1]
```diff
@@ -74,10 +74,18 @@ pub fn layout_lr(
     // Remove weak SpacingFragment immediately after the opening or immediately
     // before the closing.
     let mut index = 0;
+    let opening_exists = inner_fragments
+        .first()
+        .map_or(false, |f| f.class() == MathClass::Opening);
```
Comment (laurmaedje) [inline crates/typst-layout/src/math/lr.rs:79]: could use `is_some_and` or `.first().map(MathFragment::class) == Some(MathClass::Opening)`

File: crates/typst-layout/src/math/lr.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ pub fn layout_lr(
     }
 
     // Handle MathFragment::Variant fragments that should be scaled up.
-    for fragment in inner_fragments {
+    for fragment in &mut *inner_fragments {
```
Comment (laurmaedje) [inline crates/typst-layout/src/math/lr.rs:65]: why this change?

Comment (mkorje) [reply crates/typst-layout/src/math/lr.rs:65]: To make the borrow checker happy

**Comment by laurmaedje** [comment]
Thanks!
